planner/core/plan.go

```diff
@@ -210,17 +213,36 @@ func (p *baseLogicalPlan) storeTask(prop *property.PhysicalProperty, task task)
 	p.taskMap[string(key)] = task
 }
 
-// BuildKeyInfo implements LogicalPlan BuildKeyInfo interface.
-func (p *baseLogicalPlan) BuildKeyInfo() {
-	for _, child := range p.children {
-		child.BuildKeyInfo()
-	}
-	switch p.self.(type) {
+// HasMaxOneRow returns if the LogicalPlan will output at most one row.
+func HasMaxOneRow(p LogicalPlan, childMaxOneRow []bool) bool {
+	switch p.(type) {
 	case *LogicalLock, *LogicalLimit, *LogicalSort, *LogicalSelection, *LogicalApply, *LogicalProjection:
-		p.maxOneRow = p.children[0].MaxOneRow()
+		if len(childMaxOneRow) == 1 {
+			return childMaxOneRow[0]
+		}
 	case *LogicalMaxOneRow:
-		p.maxOneRow = true
+		return true
+	case *LogicalJoin:
+		if len(childMaxOneRow) == 2 {
+			return childMaxOneRow[0] && childMaxOneRow[1]
+		}
+	}
```

Comment:
For `LogicalAggregation`, if there is no `Group By` item, it should output at most one row?

Reply:
This case is handled in the method `BuildKeyInfo` of `LogicalAggregation`,